internal/vault/secrets/ephemeral/kvv2_secret_test.go

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package ephemeralsecrets_test
+
+import (
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-go/tfprotov6"
+	"github.com/hashicorp/terraform-plugin-testing/echoprovider"
+	"github.com/hashicorp/terraform-plugin-testing/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	"github.com/hashicorp/terraform-plugin-testing/knownvalue"
+	"github.com/hashicorp/terraform-plugin-testing/statecheck"
+	"github.com/hashicorp/terraform-plugin-testing/tfjsonpath"
+	"github.com/hashicorp/terraform-plugin-testing/tfversion"
+	"github.com/hashicorp/terraform-provider-vault/internal/providertest"
+	"github.com/hashicorp/terraform-provider-vault/testutil"
+	"testing"
+)
+
+// TestAccKVV2Secret confirms that a secret written to
+// a KV-V2 store in Vault is correctly read into the ephemeral resource
+//
+// Uses the Echo Provider to test values set in ephemeral resources
+// see documentation here for more details:
+// https://developer.hashicorp.com/terraform/plugin/testing/acceptance-tests/ephemeral-resources#using-echo-provider-in-acceptance-tests
+func TestAccKVV2Secret(t *testing.T) {
+	testutil.SkipTestAcc(t)
+
+	mount := acctest.RandomWithPrefix("kvv2-mount")
+	name := acctest.RandomWithPrefix("secret")
+	resource.UnitTest(t, resource.TestCase{
+		// Ephemeral resources are only available in 1.10 and later
+		TerraformVersionChecks: []tfversion.TerraformVersionCheck{
+			tfversion.SkipBelow(tfversion.Version1_10_0),
+		},
+		PreCheck: func() { testutil.TestAccPreCheck(t) },
+		// Include the provider we want to test
+		ProtoV5ProviderFactories: providertest.ProtoV5ProviderFactories,
+		// Include `echo` as a v6 provider from `terraform-plugin-testing`
+		ProtoV6ProviderFactories: map[string]func() (tfprotov6.ProviderServer, error){
+			"echo": echoprovider.NewProviderServer(),
+		},
+		Steps: []resource.TestStep{
+			{
+				Config: testKVV2SecretConfig(mount, name),
+				ConfigStateChecks: []statecheck.StateCheck{
+					statecheck.ExpectKnownValue("echo.test_krb", tfjsonpath.New("data").AtMapKey("password"), knownvalue.StringExact("password1")),
+				},
+			},
+		},
+	})
+}
+
+func testKVV2SecretConfig(mount, name string) string {
+	return fmt.Sprintf(`
+resource "vault_mount" "kvv2" {
+  path        = "%s"
+  type        = "kv"
+  options     = { version = "2" }
+}
+
+resource "vault_kv_secret_v2" "secret" {
+  mount                      = vault_mount.kvv2.path
+  name                       = "%s"
+  data_json_wo                  = jsonencode(
+    {
+      password       = "password1"
+    }
+  )
+  data_json_wo_version = 0
+}
+
+ephemeral "vault_kvv2_secret" "db_secret" {
```

Comment:
Should the ephemeral type be named the same as the resource that it is referencing? That might make it a bit more obvious. I see that is what is done in AWS for example.